lodash/lodash.d.ts

```diff
@@ -263,12 +263,12 @@ declare module _ {
         * @param array Array to compact.
         * @return (Array) Returns a new array of filtered values.
         **/
-        compact<T>(array: Array<T>): T[];
+        compact<T>(array?: Array<T>): T[];
```

Comment:
The way I think of it is that vanilla JavaScript can compile just fine through a compiler, right? Then why not existing code that uses a library such as lodash? I think it's more the job of TypeScript to prevent the following:

```ts
_.compact('foo');
```

We just need to make sure that argument is the right type (or null/undefined).

Basically, what I'm saying is that an argument shouldn't be marked as truly required unless you're throwing some kind of error in the function. For example:

```ts
function foo(bar: string, baz?: string) {
  if (_.isNull(bar) || _.isUndefined(bar)) {
    throw new Error('bar is null or undefined');
  }
  return bar + baz;
}
```